File: src/dotnet/HQ.Abstractions/Points/UpsertPointsV1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HQ.Abstractions.Points
{
    public class UpsertPointsV1
    {
        public class Request
        {
            public Guid? Id { get; set; }
            public Guid? StaffId { get; set; }
            public DateOnly Date { get; set; }
            public List<GetPointsV1.Point>? Points { get; set; } = new List<GetPointsV1.Point>();
        }

        public class Response
        {
            public Guid[] PointIds { get; set; } = null!;
        }
    }
}

File: src/angular/hq/src/app/models/Points/upsert-points-V1.ts
import { PlanningPoint } from './get-points-v1';

export interface upsertPointsRequestV1 {
  id?: string | null;
  date: string;
  staffId: string;
  points: PlanningPoint[];
}

export interface upsertPointsResponseV1 {
  pointIds: string[];
}
